server/index.js

```diff
@@ -58,13 +134,13 @@ app.get('/api/log', (req, res) => {
 });
 
 app.post('/api/watchlist', (req, res, next) => {
-  const { show, episodeName, season, number, image, isWatched = false } = req.body;
+  const { userId, show, episodeName, season, number, image, isWatched = false } = req.body;
```

Comment:
Instead of letting the client pick the user id in the body, you should get it from the `req.user` so that people can't add stuff to each other's watch lists. You'll need to look at all of your server code and decide which routes need to use the `req.user.userId` .